plugins/helper/batch_save.go

```diff
@@ -65,10 +69,18 @@ func (c *BatchSave) Add(slot interface{}) error {
 }
 
 func (c *BatchSave) Flush() error {
-	err := c.db.Clauses(clause.OnConflict{UpdateAll: true}).Create(c.slots.Slice(0, c.current).Interface()).Error
-	if err != nil {
-		return err
+	if c.hasPrimarykey {
```

Comment:
No, this operation should be carried out on initialization in  `NewBatchSave`, put it here would cause needless overhead for each operations.